commons/zenoh-config/src/lib.rs

```diff
@@ -283,6 +283,18 @@ validated_struct::validator! {
                     client_private_key: Option<String>,
                     client_certificate: Option<String>,
                 },
+                pub compression: #[derive(Default)]
+                /// **Experimental** compression feature.
+                /// Will compress the batches hop to hop (as opposed to end to end). May cause errors when
+                /// the batches's complexity is too high, causing the resulting compression to be bigger in
+                /// size than the MTU.
+                /// You must use zenoh's feature "transport_compression" to enable this.
+                Compression {
```

Comment:
For consistency with the rest of the config, I'd call the datatype `CompressionConf`.